tests/test_stac.py

```diff
@@ -207,6 +207,35 @@ def test_raster_footprint_geometry() -> None:
         item.validate()
 
 
+def test_create_item_from_hdf_without_xml() -> None:
+    """Test that an item can be created from an HDF file when XML is not available.
+
+    This tests the fallback to extracting metadata directly from the HDF file
+    when the accompanying XML metadata file is not present.
+    """
+    hdf_file = "MOD10A2.A2022033.h09v05.061.2022042050729.hdf"
+    source_hdf_path = test_data.get_path(f"data-files/{hdf_file}")
+
+    with TemporaryDirectory() as temporary_directory:
+        # Copy only the HDF file (not the XML) to ensure XML is not available
+        temp_hdf_path = os.path.join(temporary_directory, hdf_file)
+        shutil.copyfile(source_hdf_path, temp_hdf_path)
+
+        # Verify XML does not exist in temp directory
+        temp_xml_path = f"{temp_hdf_path}.xml"
+        assert not os.path.exists(temp_xml_path), "XML file should not exist"
+
+        # Create item from HDF only - should extract metadata from HDF
+        item = stactools.modis.stac.create_item(temp_hdf_path)
+
+        # Verify item was created with correct metadata
```

Comment:
These look like auto-gen comments, again I generally prefer to let the code speak for itself.

```suggestion
        temp_hdf_path = os.path.join(temporary_directory, hdf_file)
        shutil.copyfile(source_hdf_path, temp_hdf_path)
        temp_xml_path = f"{temp_hdf_path}.xml"
        assert not os.path.exists(temp_xml_path), "XML file should not exist"
        item = stactools.modis.stac.create_item(temp_hdf_path)
```

Reply:
done